pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -425,7 +425,7 @@ public Response downloadSegment(
   public Response downloadValidDocIds(
       @ApiParam(value = "Name of the table with type REALTIME", required = true, example = "myTable_REALTIME")
       @PathParam("tableNameWithType") String tableNameWithType,
-      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") String segmentName,
```

Comment:
I think we will need this annotation for special segment name. Same for the new added API

Reply:
@Jackie-Jiang I actually had to remove this annotation because it was causing the endpoint to fail for segment names with special characters.

Reply:
@Jackie-Jiang Are changes required here?